File: t/CS/OmrProcessingDemo/Operations/TwainOperation.cs
using Leadtools;
using Leadtools.Codecs;
using Leadtools.Twain;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace OmrProcessingDemo.Operations
{
   class TwainOperation : BusyOperation
   {
      private TwainSession twainSession;
      private string saveLocation;
      private RasterImageFormat format;

      public TwainOperation(TwainSession twnSession, string saveLocation, RasterImageFormat format)
      {
         this.twainSession = twnSession;
         this.saveLocation = saveLocation;
         this.format = format;
      }

      protected override void Run()
      {
         Progress(101, "Acquiring from scanner...");

         RasterImage image = null;// = twainSession.AcquireToImage(TwainUserInterfaceFlags.Show | TwainUserInterfaceFlags.Modal);

         SetTransferMode();

         twainSession.Acquire(TwainUserInterfaceFlags.Show | TwainUserInterfaceFlags.Modal);

         using (RasterCodecs codecs = MainForm.GetRasterCodecs())
         {
            codecs.Save(image, saveLocation, format, 0);
         }

         image.Dispose();

         base.Run();
      }

      private void SetTransferMode()
      {
         using (TwainCapability twnCap = new TwainCapability())
         {
            try
            {
               twnCap.Information.Type = TwainCapabilityType.ImageTransferMechanism;
               twnCap.Information.ContainerType = TwainContainerType.OneValue;

               twnCap.OneValueCapability.ItemType = TwainItemType.Uint16;
               twnCap.OneValueCapability.Value = (UInt16)TwainTransferMode.File;

               // Set the value of ICAP_XFERMECH (Image Transfer Mechanism) capability
               twainSession.SetCapability(twnCap, TwainSetCapabilityMode.Set);
            }
            catch (Exception ex)
            {
               throw ex;
            }
         }
      }


   }
}

File: t/VB/OmrProcessingDemo/Operations/TwainOperation.vb
Imports Leadtools
Imports Leadtools.Codecs
Imports Leadtools.Twain
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports System.Threading.Tasks

Class TwainOperation
   Inherits BusyOperation

   Private _twainSession As TwainSession
   Private _saveLocation As String
   Private _format As RasterImageFormat

   Public Sub New(ByVal twnSession As TwainSession, ByVal saveLocation As String, ByVal format As RasterImageFormat)
      _twainSession = twnSession
      _saveLocation = saveLocation
      _format = format
   End Sub

   Protected Overrides Sub Run()
      Progress(101, "Acquiring from scanner...")
      Dim image As RasterImage = Nothing
      SetTransferMode()
      _twainSession.Acquire(TwainUserInterfaceFlags.Show Or TwainUserInterfaceFlags.Modal)

      Using codecs As RasterCodecs = MainForm.GetRasterCodecs()
         codecs.Save(image, _saveLocation, _format, 0)
      End Using

      image.Dispose()
      MyBase.Run()
   End Sub

   Private Sub SetTransferMode()
      Using twnCap As TwainCapability = New TwainCapability()

         Try
            twnCap.Information.Type = TwainCapabilityType.ImageTransferMechanism
            twnCap.Information.ContainerType = TwainContainerType.OneValue
            twnCap.OneValueCapability.ItemType = TwainItemType.Uint16
            twnCap.OneValueCapability.Value = CUShort(TwainTransferMode.File)
            _twainSession.SetCapability(twnCap, TwainSetCapabilityMode.[Set])
         Catch ex As Exception
            Throw ex
         End Try
      End Using
   End Sub
End Class
